File: RUBY/zadanie_2.rb
def roman_to_decimal(roman)
    roman_numerals = {
        'I' => 1,
        'V' => 5,
        'X' => 10,
        'L' => 50,
        'C' => 100,
        'D' => 500,
        'M' => 1000
    }

    result = 0
    prev_value = 0
    roman.reverse.each_char do |char|
        value = roman_numerals[char]
        if value < prev_value
            result -= value
        else
            result += value
        end
        prev_value = value
    end
    result
end

puts roman_to_decimal("LXVI")  # Output: 66
puts roman_to_decimal("LXXXI") # Output: 81

File: JAVA/zadanie_2.java
import java.util.HashMap;
import java.util.Map;

public class Main {
    // Функция для преобразования римских чисел в десятичные
    public static int romanToDecimal(String roman) {
        // Карта, содержащая соответствие римских цифр и их десятичных значений
        Map<Character, Integer> romanNumerals = new HashMap<>();
        romanNumerals.put('I', 1);
        romanNumerals.put('V', 5);
        romanNumerals.put('X', 10);
        romanNumerals.put('L', 50);
        romanNumerals.put('C', 100);
        romanNumerals.put('D', 500);
        romanNumerals.put('M', 1000);

        int result = 0;
        int prevValue = 0;
        // Проход по римскому числу с конца
        for (int i = roman.length() - 1; i >= 0; i--) {
            int value = romanNumerals.get(roman.charAt(i));
            // Если текущее значение меньше предыдущего, вычитаем его
            if (value < prevValue) {
                result -= value;
            } else {
                result += value;
            }
            prevValue = value;
        }
        return result;
    }

    public static void main(String[] args) {
        // Примеры преобразования римских чисел в десятичные и вывод результата
        System.out.println(romanToDecimal("LXVI"));  // Вывод: 66
        System.out.println(romanToDecimal("LXXXI")); // Вывод: 81
    }
}
